### PR TITLE
[Merged by Bors] - refactor(Topology/Category): derive `Preregular` instances for `Profinite` and `Stonean` from the one on `CompHaus`

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dagurtomas** [comment]
!bench

File: Mathlib/Topology/Category/Stonean/EffectiveEpi.lean [hunk 1]
```diff
@@ -134,13 +127,4 @@ theorem effectiveEpiFamily_of_jointly_surjective
     EffectiveEpiFamily X π :=
   ((effectiveEpiFamily_tfae X π).out 2 0).mp surj
 
-open CompHaus Functor
-
-theorem _root_.CategoryTheory.EffectiveEpiFamily.toCompHaus
-    {α : Type} [Finite α] {B : Stonean.{u}}
```
Comment (dagurtomas) [inline Mathlib/Topology/Category/Stonean/EffectiveEpi.lean:140]: This theorem can now be proved by `infer_instance`. It was also apparently never used.

**Comment by dagurtomas** [comment]
!bench

**Comment by riccardobrasca** [comment]
!bench

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**